middleware/kubernetes/README.md

```diff
@@ -1,204 +1,250 @@
-	#
-	#  search ns1.svc.cluster.local svc.cluster.local cluster.local foo.com
-	#
-	# If no domains in the path produce an answer, a lookup on the bare question
-	# will be attempted.	
-	#
-	# A successful response will contain a question section with the original
-	# question, and an answer section containing the record for the question that
-	# actually had an answer.  This means that the question and answer will not
-	# match. For example:
-	#
-	#    # host -v -t a google.com
-	#    Trying "google.com.default.svc.cluster.local"
-	#    ;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 50957
-	#    ;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
-	#
-	#    ;; QUESTION SECTION:
-	#    ;google.com.default.svc.cluster.local. IN A
-	#
-	#    ;; ANSWER SECTION:
-	#    google.com.		175	IN	A	216.58.194.206
-	#
-	#
-	# NDOTS (default: 0) This provides an adjustable threshold to
-	# prevent server side lookups from triggering. If the number of dots before
-	# the first search domain is less than this number, then the search path will
-	# not executed on the server side.
-	#
-	# RESPONSE (default: SERVFAIL) RESPONSE can be either NXDOMAIN, SERVFAIL or
-	# NOERROR. This option causes coredns to return the given response instead of
-	# NXDOMAIN when the all searches in the path produce no results. Setting this
-	# to SERVFAIL or NOERROR should prevent the client from fruitlessly continuing
-	# the client side searches in the path after the server already checked them.
-	#
-	# RESOLV-CONF (default: /etc/resolv.conf) If specified, coredns uses this
-	# file to get the host's search domains. CoreDNS performs a lookup on these
-	# domains if the in-cluster search domains in the path fail to produce an
-	# answer. If not specified, the values will be read from the local resolv.conf
-	# file (i.e the resolv.conf file in the pod containing coredns).
-	#
-	# Enabling autopath causes coredns to use more memory since it needs to
-	# maintain a watch on all pods. If autopath and "pods verified" mode are
-	# both enabled, they will share the same watch. I.e. enabling both options
-	# should have an equivalent memory impact of just one.
-	autopath 0 SERVFAIL /etc/resolv.conf
-
-	# fallthrough
-	#
-	# If a query for a record in the cluster zone results in NXDOMAIN,
-	# normally that is what the response will be. However, if you specify
-	# this option, the query will instead be passed on down the middleware
-	# chain, which can include another middleware to handle the query.
-	fallthrough
-}
+	# host -t a google.com.default.svc.cluster.local.
+	google.com has address 216.58.194.206
+
+**NDOTS** (default: `0`) This provides an adjustable threshold to prevent server side lookups from triggering. If the number of dots before the first search domain is less than this number, then the search path will not executed on the server side.  When autopath is enabled with default settings, the search path is always conducted on the server side.
```

Comment:
maybe "when auto path is enabled with default settings, the search path is conducted whenever the query ends with `<namespace>.svc.<zone>`"